libs/wingcompiler/src/compile.ts

```diff
@@ -106,20 +139,21 @@ export async function compile(entrypoint: string, options: CompileOptions): Prom
   log?.("wing dir: %s", wingDir);
   const testing = options.testing ?? false;
   log?.("testing: %s", testing);
-  const tmpSynthDir = resolveSynthDir(targetdir, wingFile, options.target, testing, true);
+  const model = determineModelFromPlatforms(options.platform);
+  const tmpSynthDir = resolveSynthDir(targetdir, wingFile, model, testing, true);
   log?.("temp synth dir: %s", tmpSynthDir);
-  const synthDir = resolveSynthDir(targetdir, wingFile, options.target, testing);
+  const synthDir = resolveSynthDir(targetdir, wingFile, model, testing);
   log?.("synth dir: %s", synthDir);
   const workDir = resolve(tmpSynthDir, ".wing");
   log?.("work dir: %s", workDir);
 
   // TODO: couldn't be moved to the context's since used in utils.env(...)
   // in the future we may look for a unified approach
-  process.env["WING_TARGET"] = options.target;
+  process.env["WING_MODEL"] = model;
```

Comment:
```suggestion
  process.env["WING_TARGET"] = model;
```

Let's keep this `WING_TARGET` for now. `WING_MODEL` doesn't feel right when I am looking at this here.